test/com/google/enterprise/adaptor/fs/MockFileDelegate.java

```diff
@@ -180,21 +179,31 @@ public Path resolveDfsLink(Path doc) throws IOException {
   }
 
   @Override
-  public List<Path> enumerateDfsLinks(Path doc) throws IOException {
+  public DirectoryStream<Path> newDfsLinkStream(Path doc) throws IOException {
     MockFile file = getFile(doc);
     if (file.isDfsNamespace()) {
-      ImmutableList.Builder<Path> builder = ImmutableList.builder();
-      for (Path path : file.newDirectoryStream()) {
-        if (isDfsLink(path)) {
-          builder.add(path);
-        }
-      }
-      return builder.build();
+      return new MockDfsLinkDirectoryStream(file);
     } else {
       throw new IOException("Not a DFS Root: " + doc);
     }
   }
 
+  private class MockDfsLinkDirectoryStream extends AbstractPathDirectoryStream {
+    public MockDfsLinkDirectoryStream(MockFile namespace) throws IOException {
+      paths = Iterables.filter(namespace.newDirectoryStream(),
```

Comment:
For some reason, your comment was lost here (but it does show up in the email).

John says:
This is cute, but longer, harder to read, and inconsistent with MockDirectoryStream. The only benefit I can see is that it could be used as a direct argument to super(), if paths is made final in the superclass. Even there, a simple helper method or just removing this class and putting this code into newDfsLinkStream seems better

Brett responds:
So I restored the use of ImmutableList.Builder and moved the implementation into newDfsLinkStream above.